### PR TITLE
Allow timeouts, add some basic tests

### PR DESCRIPTION
**Pull request by goodspark**
~Note: 3.7 is tested, but since the tests are not comprehensive, them
passing is not a good indication that 3.7 should be supported (ie. setup.py
updated). So I haven't added it to the classifiers section.~ 3.7 doesn't seem
to be stable on Travis. Ignoring for now.

Also adds Travis YML file to run tests.

Addresses #11

### CODE REVIEW

File: yelpapi/yelpapi.py [hunk 1]
```diff
@@ -71,11 +71,20 @@ class YelpAPIError(Exception):
...
-            Instantiate a YelpAPI object. An API key from Yelp is required. 
+            Instantiate a YelpAPI object. An API key from Yelp is required.
+
+            required parameters:
+                * api_key - Our Yelp API key
+
+            optional parameters:
+                * timeout_s - Timeout, in seconds, to set for all API calls. If the
+                  the timeout expires before the request completes, then a Timeout
+                  exception will be raised.
```
Comment (emfloyd2) [inline yelpapi/yelpapi.py]: Might be worth adding a note saying the default is to wait forever

File: tests/test_yelpapi.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+        with pytest.raises(YelpAPI.YelpAPIError) as e:
+            yelp._query(url, **random_dict)
+
+        assert 1 == mock_call.call_count
+        exc = e.value
+        assert '{}: {}'.format(error_code, error_description) == exc.args[0]
+
+
+class TestSearch:
```
[the remark this answers is not in the document]
Comment (goodspark) [reply tests/test_yelpapi.py]: It's testing `search_query` though?

File: CHANGES.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changes
 
+## TBD
```
Comment (gfairchild) [inline CHANGES.md]: Let's just call this version 2.3. We'll also need to update the version number in `setup.py`.

File: tests/test_yelpapi.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gfairchild) [reply tests/test_yelpapi.py]: I could go either way. I could also see renaming `search_query` to `business_search_query`.

**Comment by gfairchild** [comment]
One other thought: structurally, wouldn't it make more sense for `TestSearch` (or `TestSearchQuery`) to be a subclass within `TestYelpAPI`?

**Comment by goodspark** [comment]
Yeah, I just personally don't like excessive indenting. But not a hard point for me. Can move it.

**Comment by gfairchild** [comment]
Yeah, I think I’d prefer that if you don’t mind. Structurally, it just makes more sense to me.

**Pull request merged**